envoyauth/response.go

```diff
@@ -67,12 +67,11 @@ func NewEvalResult(opts ...func(*EvalResult)) (*EvalResult, StopFunc, error) {
 func (result *EvalResult) GetTxn(ctx context.Context, store storage.Store) (storage.Transaction, TransactionCloser, error) {
 	params := storage.TransactionParams{}
 
-	noopCloser := func(ctx context.Context, err error) error {
-		return nil // no-op default
-	}
-
 	txn, err := store.NewTransaction(ctx, params)
 	if err != nil {
+		noopCloser := func(ctx context.Context, err error) error {
```

Comment:
nit
```suggestion
		noopCloser := func(context.Context, error) error {
```

Reply:
Perhaps this function literal is simple enough that it won't leak to the heap, but you could try to rewrite it a package level function, e.g.

```
func noopCloser(ctx context.Context, err error) error {
    return nil // no-op default
}
```

and then reference that instead. Measure and see if it makes any difference :)

Reply:
I've made this change. However, there are no significant changes from the benchmark perspective.